mithril-aggregator/src/runtime/runner.rs

```diff
@@ -460,6 +460,8 @@ impl AggregatorRunnerTrait for AggregatorRunner {
     }
 
     async fn inform_new_epoch(&self, epoch: Epoch) -> StdResult<()> {
+        self.dependencies.upkeep_service.run().await?;
```

Comment:
I think it would probably be better to extract that into a new function.
Also, this new function should be the last one called before reopening the signer registration (sorry I made a typo in my previous comment where I meant 'before' where I wrote 'after').